### PR TITLE
Enabled ENUM sensors

### LINKED ISSUE #6: HMIP-SWDO OPEN/CLOSED not shown
I am using HMIP-SWDO and want to see window opening status.
Any idea how to troubleshoot why a lot of metrics of the HMIP-SWDO beside the intresting opening state? 

I guess it has something todo with OPEN/CLOSED are strings and have to be converted to int 1/0 in the script?

```
{'ADDRESS': '0000DA499C6BE4',
  'AES_ACTIVE': 1,
  'AVAILABLE_FIRMWARE': '0.0.0',
  'CHILDREN': ['0000DA499C6BE4:0', '0000DA499C6BE4:1', '0000DA499C6BE4:2'],
  'DIRECTION': 0,
  'FIRMWARE': '1.16.8',
  'FIRMWARE_UPDATE_STATE': 'UP_TO_DATE',
  'FLAGS': 1,
  'GROUP': '',
  'INDEX': 0,
  'INTERFACE': '',
  'LINK_SOURCE_ROLES': '',
  'LINK_TARGET_ROLES': '',
  'PARAMSETS': ['MASTER', 'SERVICE'],
  'PARENT': '',
  'PARENT_TYPE': '',
  'RF_ADDRESS': 8415741,
  'ROAMING': 0,
  'RX_MODE': 4,
  'SUBTYPE': 'SWD',
  'TEAM': '',
  'TEAM_CHANNELS': [],
  'TEAM_TAG': '',
  'TYPE': 'HMIP-SWDO',
  'UPDATABLE': True,
  'VERSION': 2},
{'ADDRESS': '0000DA499C6BE4:0',
  'AES_ACTIVE': 1,
  'AVAILABLE_FIRMWARE': '',
  'CHILDREN': [],
  'DIRECTION': 0,
  'FIRMWARE': '',
  'FIRMWARE_UPDATE_STATE': '',
  'FLAGS': 1,
  'GROUP': '',
  'INDEX': 0,
  'INTERFACE': '',
  'LINK_SOURCE_ROLES': '',
  'LINK_TARGET_ROLES': '',
  'PARAMSETS': ['MASTER', 'VALUES', 'SERVICE'],
  'PARENT': '0000DA499C6BE4',
  'PARENT_TYPE': 'HMIP-SWDO',
  'RF_ADDRESS': 0,
  'ROAMING': 0,
  'RX_MODE': 0,
  'SUBTYPE': '',
  'TEAM': '',
  'TEAM_CHANNELS': [],
  'TEAM_TAG': '',
  'TYPE': 'MAINTENANCE',
  'UPDATABLE': True,
  'VERSION': 2},
{'ADDRESS': '0000DA499C6BE4:1',
  'AES_ACTIVE': 1,
  'AVAILABLE_FIRMWARE': '',
  'CHILDREN': [],
  'DIRECTION': 1,
  'FIRMWARE': '',
  'FIRMWARE_UPDATE_STATE': '',
  'FLAGS': 1,
  'GROUP': '',
  'INDEX': 1,
  'INTERFACE': '',
  'LINK_SOURCE_ROLES': 'WINDOW_SWITCH CONDITIONAL_SWITCH',
  'LINK_TARGET_ROLES': '',
  'PARAMSETS': ['MASTER', 'VALUES', 'LINK', 'SERVICE'],
  'PARENT': '0000DA499C6BE4',
  'PARENT_TYPE': 'HMIP-SWDO',
  'RF_ADDRESS': 0,
  'ROAMING': 0,
  'RX_MODE': 0,
  'SUBTYPE': '',
  'TEAM': '',
  'TEAM_CHANNELS': [],
  'TEAM_TAG': '',
  'TYPE': 'SHUTTER_CONTACT',
  'UPDATABLE': True,
  'VERSION': 2},
 {'ADDRESS': '0000DA499C6BE4:2',
  'AES_ACTIVE': 1,
  'AVAILABLE_FIRMWARE': '',
  'CHILDREN': [],
  'DIRECTION': 0,
  'FIRMWARE': '',
  'FIRMWARE_UPDATE_STATE': '',
  'FLAGS': 0,
  'GROUP': '',
  'INDEX': 2,
  'INTERFACE': '',
  'LINK_SOURCE_ROLES': '',
  'LINK_TARGET_ROLES': '',
  'PARAMSETS': ['MASTER', 'VALUES', 'SERVICE'],
  'PARENT': '0000DA499C6BE4',
  'PARENT_TYPE': 'HMIP-SWDO',
  'RF_ADDRESS': 0,
  'ROAMING': 0,
  'RX_MODE': 0,
  'SUBTYPE': '',
  'TEAM': '',
  'TEAM_CHANNELS': [],
  'TEAM_TAG': '',
  'TYPE': 'ALARM_COND_SWITCH_TRANSMITTER',
  'UPDATABLE': True,
  'VERSION': 2},
```

Parameter dump:
```
--dump_parameters 0000DA499C6BE1:1
{'STATE': 0}
```

### PR DESCRIPTION
**Pull request by kremers**
Signed-off-by: Martin Kremers <info@martinkremers.de>

Fixes #6

### CODE REVIEW

**Comment by sfudeus** [comment]
Hi Martin - thanks for your contribution. 
Does your proposed patch really solve your problem? Adding the enumset as a label seems a bit awkward to me. What is the metric value in that case? Is that supposed to be the index in the enumset? 
Unfortunately I have no device here which would show an enum different then index 0 currently. 

Regarding the labels of the metrics, I'd sooner add a label with the name of the enum item and value 1
Example:
Instead of 
```
homematic_current_status{ccu="hostname",device="xxxxxxxxxxxx:6",device_type="ENERGIE_METER_TRANSMITTER",mapped_name="xxxxxxxxxx:6",param_desc="['NORMAL', 'OVERFLOW', 'UNDERFLOW']",parent_device_type="HMIP-PSM"} 0.0
```
it would be
```
homematic_current_status{ccu="hostname",device="xxxxxxxxxxxx:6",device_type="ENERGIE_METER_TRANSMITTER",mapped_name="xxxxxxxxxx:6",status="NORMAL",parent_device_type="HMIP-PSM"} 1.0
homematic_current_status{ccu="hostname",device="xxxxxxxxxxxx:6",device_type="ENERGIE_METER_TRANSMITTER",mapped_name="xxxxxxxxxx:6",status="OVERFLOW",parent_device_type="HMIP-PSM"} 0.0
homematic_current_status{ccu="hostname",device="xxxxxxxxxxxx:6",device_type="ENERGIE_METER_TRANSMITTER",mapped_name="xxxxxxxxxx:6",status="UNDERFLOW",parent_device_type="HMIP-PSM"} 0.0
```

What do you think?

File: exporter.py [hunk 1]
```diff
@@ -96,7 +96,14 @@ def generateMetrics(self):
               paramDesc = paramsetDescription.get(key)
               paramType = paramDesc.get('TYPE')
               if paramType in ['FLOAT', 'INTEGER', 'BOOL']:
-                self.processSingleValue(devAddress, devType, devParentAddress, devParentType, paramType, key, paramset.get(key))
+                self.processSingleValue(devAddress, devType, devParentAddress, devParentType, paramType, key, paramset.get(key), paramDesc.get('TYPE'))
+              elif paramType == 'ENUM':
+                logging.debug("Found {}: desc: {} key: {}".format(paramType,paramDesc,paramset.get(key)))
+                self.processSingleValue(devAddress, devType, devParentAddress, devParentType, paramType, key, paramset.get(key), paramDesc.get('VALUE_LIST'))
+              else:
+                  # ATM Unsupported like HEATING_CONTROL_HMIP.PARTY_TIME_START,
```
Comment (sfudeus) [inline exporter.py]: wrong indentation

**Comment by kremers** [comment]
> Does your proposed patch really solve your problem?

Yes, I am using it already with an own docker image on my rpi

> Is that supposed to be the index in the enumset?

Yes

> What do you think?

Having all enum values and the selected index will allow queries to be way simpler using enum/value mapping in grafana later on. One con is that it won't adapt well for changing enums, but I consider that a very rare case. 

As far as I understand StateSetMetricFamily has to be used to generate your intended result?

File: exporter.py (hunk 1 above)
Comment (kremers) [reply exporter.py]: done

**Comment by kremers** [comment]
Alternative Version for review:  #8

**Comment by kremers** [comment]
Closed #8 in favour of the most up to date master, container your suggestions

**Comment by sfudeus** [comment]
Sorry for the delay, looks good, thanks.

**Pull request merged**